src/Microsoft.PowerShell.SHiPS/Utility/SHiPSExtensions.cs

```diff
@@ -137,6 +137,15 @@ internal static string TrimDrive(this string path, SHiPSDrive drive)
 
             return path1.TrimEnd('\\', '/');
         }
+
+        internal static object HomePath(this string name, SHiPSProvider provider)
+        {
+            if(string.IsNullOrWhiteSpace(name) || provider == null) { return null;}
+
+            var command = "Get-Variable {0}".StringFormat(name);
```

Comment:
Please add a TODO here as a reminder that we need to optimize this action using OSPlatform or some other means